sentry-core/src/main/java/io/sentry/core/EnvelopeSender.java

```diff
@@ -144,13 +146,18 @@ private void processEnvelope(final @NotNull SentryEnvelope envelope, final @Null
                     SentryLevel.WARNING,
                     "Timed out waiting for event submission: %s",
                     event.getEventId());
+
+                if (hint instanceof Retryable) {
```

Comment:
H: Not sure about this. It could mean we're capturing events twice if we mark the event as retry=true and it actually finishes being sent. Why is this change introduced? We don't seem to have an issue missing events, right?

Reply:
it's a timed out and not a known exception that returns immediately which is 99.9% of the cases (no connection or something), so it really didn't have time enough due to different things, eg, long queue, very slow connection and low-end device, taking more time to serialize/deserialize, etc, I did reproduce this using emulator and very bad configurations purposely

Reply:
It times out waiting for the synchronization mechanism to be called as opposed to an HTTP request timeout. It's very possible that we'll give up waiting here before the HTTP request times out (unless we guarantee that the HTTP request times out before this `waitFlush` method here.

It's also an issue where have some bug in the SDK and now we're keeping things forever because this method is never called.

Reply:
I'm not talking about only http request timeout, it's altogether.

Reply:
I removed those `Retryable` checks from this PR and can come up with a new PR, merging this one without them